test/Microsoft.AspNetCore.Server.KestrelTests/RequestTargetProcessingTests.cs

```diff
@@ -0,0 +1,131 @@
+                await context.Response.WriteAsync("Hello World");
+            }, testContext, "http://127.0.0.1/\u0041\u030A"))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.SendEnd(
+                        "GET /%41%CC%8A/A/../B/%41%CC%8A HTTP/1.0",
+                        "",
+                        "");
+                    await connection.ReceiveEnd(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {testContext.DateHeaderValue}",
+                        "Content-Length: 11",
+                        "",
+                        "Hello World");
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("/")]
+        [InlineData("/.")]
+        [InlineData("/..")]
+        [InlineData("/./.")]
+        [InlineData("/./..")]
+        [InlineData("/../.")]
+        [InlineData("/../..")]
+        [InlineData("/path")]
+        [InlineData("/path?foo=1&bar=2")]
+        [InlineData("/hello%20world")]
+        [InlineData("/hello%20world?foo=1&bar=2")]
+        [InlineData("/base/path")]
+        [InlineData("/base/path?foo=1&bar=2")]
+        [InlineData("/base/hello%20world")]
+        [InlineData("/base/hello%20world?foo=1&bar=2")]
+        public async Task RequestFeatureContainsRawTarget(string requestTarget)
+        {
+            var testContext = new TestServiceContext();
+
+            using (var server = new TestServer(async context =>
+            {
+                Assert.Equal(requestTarget, context.Features.Get<IHttpRequestFeature>().RawTarget);
+
+                context.Response.Headers["Content-Length"] = new[] { "11" };
+                await context.Response.WriteAsync("Hello World");
+            }, testContext))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.SendEnd(
+                        $"GET {requestTarget} HTTP/1.0",
+                        "",
+                        "");
+                    await connection.ReceiveEnd(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {testContext.DateHeaderValue}",
+                        "Content-Length: 11",
+                        "",
+                        "Hello World");
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("*")]
+        [InlineData("*/?arg=value")]
+        [InlineData("*?arg=value")]
+        [InlineData("DoesNotStartWith/")]
+        [InlineData("DoesNotStartWith/?arg=value")]
+        [InlineData("DoesNotStartWithSlash?arg=value")]
+        [InlineData("./")]
+        [InlineData("../")]
+        [InlineData("../.")]
+        [InlineData(".././")]
+        [InlineData("../..")]
+        [InlineData("../../")]
+        public async Task NonPathRequestTargetSetInRawTarget(string requestTarget)
```

Comment:
WebListener actually rejects those requests. Maybe we should do the same in Kestrel? The only request I was able to make to a target not starting with `/` was `OPTIONS *`.